### PR TITLE
Issue/2345

### PR DESCRIPTION
**Pull request by xsrust**
Addresses #2345

Overview of changes:
- Make `Download Global Usage` super-admin only
- Create `Download Monthly Usage` button to get a CSV of:
   - `Month, new plans, new users, downloads, plans shared`  
   - For each month since creation
- New models/services for statistics on exported and shared plans
  - Added these to the stats rake tasks
- Add parallel gem
  - Add parallelized tasks for `rake stat:build` and `rake stat:build_last_month`
  - Added threading to the other `create_` services
  - Change DB default pooling to 16

Note:
UI with the multiple buttons looks funny at the moment, but I will refactor the layout in another feature's PR as I'd started this work there.

### CODE REVIEW

File: config/database.yml.sample [hunk 1]
```diff
@@ -3,7 +3,7 @@ defaults: &defaults
   encoding: <%= ENV['DB_ADAPTER'] == "mysql2" ? "utf8mb4" : "" %>
   username: <%= ENV["DB_ADAPTER"] == "postgresql" ? 'postgres' : '' %>
   database: roadmap_<%= ENV['RAILS_ENV'] %>
-  pool: 5
+  pool: 16
```
Comment (briri) [inline config/database.yml.sample]: Is this for the parallel gem?

File: lib/tasks/stat.rake [hunk 2]
```diff
@@ -4,8 +4,51 @@ namespace :stat do
...
+    tasks = ["stat:create:created_plan",
+             "stat:create:joined_user",
+             "stat:create:shared_plan",
+             "stat:create:exported_plan",
+             "stat:create_last_month:created_plan",
+             "stat:create_last_month:joined_user",
+             "stat:create_last_month:shared_plan",
+             "stat:create_last_month:exported_plan"]
+
+      Parallel.each(tasks, progress: "Building Stats", in_processes: 4) do |task|
```
Comment (briri) [inline lib/tasks/stat.rake]: 👍 this is great

**Comment by briri** [comment]
looks like you have some failing stats tests too that need to be addressed

File: config/database.yml.sample (hunk 1 above)
Comment (xsrust) [reply config/database.yml.sample]: yeah, essentially each thread/process takes an additional db connection, so it was flooding the default 5.  16 is a bit overkill for development, but it means we don't end up with connection-timeout errors

**Comment by xsrust** [comment]
Looks like the github-workflows are getting confused when checking things out from the DCC fork.
I've updated the `mySQL` side of our workflows to the `checkout@v2` to patch the problem.  
If that works, I'll apply the same fix to the remaining workflow files.

https://github.com/actions/checkout/issues/23

**Comment by xsrust** [comment]
Going to merge this so I can rebase the work on https://github.com/DMPRoadmap/roadmap/issues/1682 
They both affect the services and usage-dashboard heavily, so there's likely to be conflicts to fix on that branch.

**Pull request merged**